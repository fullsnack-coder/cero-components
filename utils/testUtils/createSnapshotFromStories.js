import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
