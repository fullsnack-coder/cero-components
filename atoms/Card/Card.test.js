import * as stories from './Card.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Card ]', () => {
  describe('#Card', () => createSnapshotFromStories('Card', stories))
})
