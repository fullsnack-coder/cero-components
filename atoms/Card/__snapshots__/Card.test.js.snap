// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Card ] #Card when 'Card' Clickable is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="card color-base size-sm is-clickable"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</DocumentFragment>
`;

exports[`[ atoms / Card ] #Card when 'Card' Colors is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="card color-base size-sm"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    class="card color-primary size-sm"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    class="card color-secondary size-sm"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</DocumentFragment>
`;

exports[`[ atoms / Card ] #Card when 'Card' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="card color-base size-sm"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</DocumentFragment>
`;

exports[`[ atoms / Card ] #Card when 'Card' Dragabble is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="card color-base size-sm is-draggable"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</DocumentFragment>
`;

exports[`[ atoms / Card ] #Card when 'Card' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="card color-base size-sm"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    class="card color-base size-md"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
  <div
    class="card color-base size-lg"
  >
    Lacerations coaster sort comings windlance happily EIf-witch handful unbefitting? Decide rising startled Aragorn invitations midnight deserves fortunes innards. You cannot hide. I see you. There is no life in the void. Only death. Mirror Emyn dreamed!
  </div>
</DocumentFragment>
`;
